src/sql/src/parsers/alter_parser.rs

```diff
@@ -143,12 +152,41 @@ impl ParserContext<'_> {
                 .parse_identifier(false)
                 .context(error::SyntaxSnafu)?,
         );
-        let target_type = self.parser.parse_data_type().context(error::SyntaxSnafu)?;
 
-        Ok(AlterTableOperation::ChangeColumnType {
-            column_name,
-            target_type,
-        })
+        if self.parser.parse_keyword(Keyword::SET) {
+            self.parser
+                .expect_keyword(Keyword::FULLTEXT)
+                .context(error::SyntaxSnafu)?;
+
+            let options = self
+                .parser
+                .parse_options(Keyword::WITH)
+                .context(error::SyntaxSnafu)?
+                .into_iter()
+                .map(parse_option_string)
+                .collect::<Result<HashMap<String, String>>>()?;
+
+            for key in options.keys() {
+                ensure!(
+                    validate_column_fulltext_alter_option(key),
```

Comment:
`key.to_ascii_lowercase ()` ?

Reply:
the function `parse_option_string` has converted it to lower case already, so I removed it.